examples/owneredit.py

```diff
@@ -0,0 +1,517 @@
+                self.ldap_session.search(self.domain_dumper.root, '(sAMAccountName=%s)' % escape_filter_chars(_lookedup_owner), attributes=['objectSid'])
+            elif self.new_owner_DN is not None:
+                _lookedup_owner = self.new_owner_DN
+                self.ldap_session.search(self.domain_dumper.root, '(distinguishedName=%s)' % _lookedup_owner, attributes=['objectSid'])
+            try:
+                self.new_owner_SID = format_sid(self.ldap_session.entries[0]['objectSid'].raw_values[0])
+                logging.debug("Found new owner SID: %s" % self.new_owner_SID)
+            except IndexError:
+                logging.error('New owner SID not found in LDAP (%s)' % _lookedup_owner)
+                exit(1)
+
+    def read(self):
+        current_owner_SID = format_sid(self.target_principal_security_descriptor['OwnerSid']).formatCanonical()
+        logging.info("Current owner information below")
+        logging.info("- SID: %s" % current_owner_SID)
+        logging.info("- sAMAccountName: %s" % self.resolveSID(current_owner_SID))
+        self.ldap_session.search(self.domain_dumper.root, '(objectSid=%s)' % current_owner_SID, attributes=['distinguishedName'])
+        current_owner_distinguished_name = self.ldap_session.entries[0]
+        logging.info("- distinguishedName: %s" % current_owner_distinguished_name['distinguishedName'])
+
+    def write(self):
+        logging.debug('Attempt to modify the OwnerSid')
+        _new_owner_SID = ldaptypes.LDAP_SID()
+        _new_owner_SID.fromCanonical(self.new_owner_SID)
+        # lib doesn't set this, but I don't known if it's needed
+        # _new_owner_SID['SubLen'] = len(_new_owner_SID['SubAuthority'])
+        self.target_principal_security_descriptor['OwnerSid'] = _new_owner_SID
+
+        self.ldap_session.modify(
+            self.target_principal.entry_dn,
+            {'nTSecurityDescriptor': (ldap3.MODIFY_REPLACE, [
+                self.target_principal_security_descriptor.getData()
+            ])},
+            controls=security_descriptor_control(sdflags=0x01))
+        if self.ldap_session.result['result'] == 0:
+            logging.info('OwnerSid modified successfully!')
+        else:
+            if self.ldap_session.result['result'] == 50:
+                logging.error('Could not modify object, the server reports insufficient rights: %s',
+                              self.ldap_session.result['message'])
+            elif self.ldap_session.result['result'] == 19:
+                logging.error('Could not modify object, the server reports a constrained violation: %s',
+                              self.ldap_session.result['message'])
+            else:
+                logging.error('The server returned an error: %s', self.ldap_session.result['message'])
+
+    # Attempts to retrieve the Security Descriptor of the specified target
+    def search_target_principal_security_descriptor(self):
```

Comment:
move duplicated function ( and resolveSID also ) to a lib ( i.e. impacket/examples/ldap_helpers.py ) to be used by dacledit.py and this one